modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java

```diff
@@ -398,9 +398,29 @@ public OperationsMap postProcessOperationsWithModels(OperationsMap objs, List<Mo
             }
         }
 
+        this.addConditionalImportInformation(objs);
+
         return objs;
     }
 
+    private void addConditionalImportInformation(OperationsMap operations) {
+        boolean hasPathParams = false;
+        boolean hasBodyParams = false;
+
+        for (CodegenOperation op : operations.getOperations().getOperation()) {
+            if (op.getHasPathParams()) {
+                hasPathParams = true;
+            }
+            if (op.getHasBodyParam()) {
+                hasBodyParams = true;
+            }
+        }
+
+        additionalProperties.put("hasPathParams", hasPathParams);
+        additionalProperties.put("hasBodyParams", hasBodyParams);
```

Comment:
thanks for the PR. does this fix work for APIs with different tags (e.g. Pet, User) ? looks like if `Pet` APIs have an endpoint with body parameter while `Users` APIs doesn't have any body parameter, `hasBodyParams` is also set to true for `Users` APIs (operations).

Reply:
I actually added an open api spec and a sample generated code that creates 3 different api files with their own tags (api_none.go, api_body.go, api_path.go) to test 3 different cases, I haven't make a case for "both" but the logic is independent for both case and just a combination of body and path tag. 

Each of them has the imports according to what they need to use.

Reply:
That's good. I'll do some tests later to confirm